component/discovery/kubernetes/k8s.go

```diff
@@ -0,0 +1,140 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/component/metrics/scrape"
+	promk8s "github.com/prometheus/prometheus/discovery/kubernetes"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.k8s",
+		Args:    SDConfig{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(SDConfig))
+		},
+	})
+}
+
+// SDConfig is a conversion of discover/kubernetes/SDConfig to be compatible with flow
+type SDConfig struct {
+	APIServer          config.URL              `river:"api_server,attr,optional"`
+	Role               string                  `river:"role,attr"`
+	KubeConfig         string                  `river:"kubeconfig_file,attr,optional"`
+	HTTPClientConfig   config.HTTPClientConfig `river:"http_client_config,attr,optional"`
+	NamespaceDiscovery NamespaceDiscovery      `river:"namespaces,attr,optional"`
+	Selectors          []SelectorConfig        `river:"selectors,attr,optional"`
```

Comment:
All three of these should be blocks and not attributes; blocks are for grouping related settings together while attributes are for values which can be passed around as a whole.